action.yml

```diff
@@ -0,0 +1,129 @@
+    description: 'Docker build context'
+    required: false
+    default: '.'
+  use_distgen:
+    description: 'The action will use distgen for generating dockerfiles if true'
+    required: false
+    default: 'false'
+
+runs:
+  using: "composite"
+  steps:
+    - name: Check if Action is run from sclorg repository
+      if: github.repository_owner != 'sclorg'
+      shell: bash
+      run: |
+        echo "This action can be run only for sclorg repositories"
+        exit 1
+
+    - name: Checkout git
+      uses: actions/checkout@v2
+      with:
+       submodules: true
+
+    - name: Get base image name
+      id: base-image-name
+      shell: bash
+      run: |
+        # This command returns row with BASE_IMAGE_NAME
+        row=$(grep "BASE_IMAGE_NAME" Makefile)
+        # Return only base image name
+        BASE_IMAGE_ROW=${row/BASE_IMAGE_NAME = /}
+        echo ::set-output name=image_name::$BASE_IMAGE_ROW
+
+    - name: Get short version
+      id: short_version
+      shell: bash
+      run: |
+        ver="${{ inputs.version }}"
+        echo "::set-output name=SHORT_VER::${ver//./}"
+
+    - name: Check if .exclude-${{ inputs.tag }} is present in version directory
+      id: check_exclude_file
+      # https://github.com/marketplace/actions/file-existence
+      uses: andstor/file-existence-action@v1
+      with:
+        files: "${{ inputs.version }}/.exclude-${{ inputs.tag }}"
+
+    - name: Login to registry
+      if: steps.check_exclude_file.outputs.files_exists == 'false'
+      uses: redhat-actions/podman-login@v1
+      with:
+        registry: ${{ inputs.registry }}
+        username: ${{ inputs.registry_username }}
+        password: ${{ inputs.registry_token }}
+
+    - name: Install distgen and generate source
+      if: ${{ inputs.use_distgen }} == 'true' && steps.check_exclude_file.outputs.files_exists == 'false'
+      shell: bash
+      run: |
+        sudo apt-get update -y && sudo apt-get install -y python3 python3-pip
+        pip3 -v install distgen
+        DG=$HOME/.local/bin/dg make generate-all
+
+    - name: Check if ${{ inputs.dockerfile }} is present in version directory
+      if: steps.check_exclude_file.outputs.files_exists == 'false'
+      id: check_dockerfile_file
+      # https://github.com/marketplace/actions/file-existence
+      uses: andstor/file-existence-action@v1
+      with:
+        files: "${{ inputs.version }}/${{ inputs.dockerfile }}"
+        allow_failure: true # fails the Action if Dockerfile is missing
+
+    - name: Build image
+      if: steps.check_exclude_file.outputs.files_exists == 'false'
+      id: build-image
+      # https://github.com/marketplace/actions/buildah-build
+      uses: redhat-actions/buildah-build@v2
+      with:
+        dockerfiles: ${{ inputs.version }}/${{ inputs.dockerfile }}
+        image: ${{ steps.base-image-name.outputs.image_name}}-${{ steps.short_version.outputs.SHORT_VER }}-${{ inputs.tag }}
+        context: ${{ inputs.docker_context }}
```

Comment:
This should be generated. ${{ inputs.docker_context }} can be empty. I don't know what will be behavior after.

Reply:
The default value of `inputs.docker_context` is ".".  This variable will be empty only if the user explicitly inserts an empty string into the docker_context input. I do not see it as a problem, as we do not check any other user input for validity either.

Reply:
I see, I don't catch the default value. You are right.